packages/jellyfish-api-core/__tests__/category/icxorderbook/makeOffer.test.ts

```diff
@@ -0,0 +1,467 @@
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { ICXGenericResult, ICXOfferInfo, ICXOrderInfo, ICXOffer, ICXOrder, ICXOrderStatus, ICXOrderType, UTXO } from '../../../src/category/icxorderbook'
+import BigNumber from 'bignumber.js'
+import { ICXSetup, symbolDFI, accountDFI, idDFI, accountBTC, ICX_TAKERFEE_PER_BTC, DEX_DFI_PER_BTC_RATE } from './icx_setup'
+import { RpcApiError } from '@defichain/jellyfish-api-core'
+
+describe('ICXOrderBook.makeOffer', () => {
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+  const icxSetup = new ICXSetup(container)
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+    await icxSetup.createAccounts()
+    await icxSetup.createBTCToken()
+    await icxSetup.initializeTokensIds()
+    await icxSetup.mintBTCtoken(100)
+    await icxSetup.fundAccount(accountDFI, symbolDFI, 500)
+    await icxSetup.fundAccount(accountBTC, symbolDFI, 10) // for fee
+    await icxSetup.createBTCDFIPool()
+    await icxSetup.addLiquidityToBTCDFIPool(1, 100)
+    await icxSetup.setTakerFee(0.001)
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should make an partial offer to an sell DFI order', async () => {
+    const accountDFIBeforeOrder = await container.call('getaccount', [accountDFI, {}, true])
+    // create order - maker
+    const order: ICXOrder = {
+      tokenFrom: idDFI,
+      chainTo: 'BTC',
+      ownerAddress: accountDFI,
+      receivePubkey: '037f9563f30c609b19fd435a19b8bde7d6db703012ba1aba72e9f42a87366d1941',
+      amountFrom: new BigNumber(15),
+      orderPrice: new BigNumber(0.01)
+    }
+    let result: ICXGenericResult = await client.icxorderbook.createOrder(order, [])
+    const createOrderTxId = result.txid
+    await container.generate(1)
+
+    // list ICX orders
+    let orders: Record<string, ICXOrderInfo | ICXOfferInfo> = await client.call('icx_listorders', [], 'bignumber')
```

Comment:
```suggestion
    const orders: Record<string, ICXOrderInfo | ICXOfferInfo> = await client.call('icx_listorders', [], 'bignumber')
```

avoid using let, the reason behind this is for testing is it allows a better reading experience where we can read and forget the variable.

When encountering a `let`, it usually mutates and has to be tracked over a sequence of mutating events. Making the code review tiring experience tiring as cognitively I need to remember all the mutating variables.

> It's okay and preferred to use a long and explicit variable name in tests.